lambdas/validate-oauth-callback/src/main/java/uk/gov/di/ipv/core/validateoauthcallback/ValidateOAuthCallbackHandler.java

```diff
@@ -111,20 +112,27 @@ public Map<String, Object> handleRequest(CriCallbackRequest callbackRequest, Con
             String ipvSessionId = callbackRequest.getIpvSessionId();
             String criOAuthSessionId = callbackRequest.getState();
 
-            if (ipvSessionId != null) {
+            if (ipvSessionId != null && !ipvSessionId.isEmpty()) {
                 ipvSessionItem = ipvSessionService.getIpvSession(ipvSessionId);
-            } else if (criOAuthSessionId == null) {
+            } else if (criOAuthSessionId != null && !criOAuthSessionId.isEmpty()) {
+                criOAuthSessionItem =
+                        criOAuthSessionService.getCriOauthSessionItem(criOAuthSessionId);
+                var mapMessage =
+                        new StringMapMessage()
+                                .with("message", "No ipvSession for existing CriOAuthSession")
+                                .with("criId", criOAuthSessionItem.getCriId())
+                                .with("criOAuthSessionId", criOAuthSessionId);
+                LOGGER.info(mapMessage);
+                Map<String, Object> pageOutput =
+                        StepFunctionHelpers.generatePageOutputMap(
+                                "error",
+                                HttpStatus.SC_UNAUTHORIZED,
+                                PYI_TIMEOUT_RECOVERABLE_PAGE_ID);
+                pageOutput.put("criOAuthSessionId", criOAuthSessionId);
+                return pageOutput;
+            } else {
                 throw new HttpResponseExceptionWithErrorBody(
                         HttpStatus.SC_BAD_REQUEST, ErrorResponse.MISSING_OAUTH_STATE);
-            } else {
-                ipvSessionItem =
-                        ipvSessionService
-                                .getIpvSessionByCriOAuthSessionId(criOAuthSessionId)
```

Comment:
I'm probably behind on this but why do we no longer want to do this?

Reply:
We shouldn't really be looking up the ipvSession based on the criOAuthSession - although we technically can as they are relational in dynamo, I think @kerrr suggested it was unnecessary / insecure for this specific issue